pkg/api/validation/validation.go

```diff
@@ -429,6 +430,17 @@ func ValidateLabels(labels map[string]string, field string) errs.ValidationError
 	return allErrs
 }
 
+// ValidateAnnotations validates that a set of annotations are correctly defined.
+func ValidateAnnotations(annotations map[string]string, field string) errs.ValidationErrorList {
```

Comment:
I cringed at the duplication, too.  I removed it and reused the ValidationLabels func.